WrapImage/nifti_wrapper.py

```diff
@@ -58,7 +61,20 @@ def save_nifti_3d(data, output_file):
         bvals = read_json_file(bval_file)
 
         # Pass additional arguments to the algorithm
+        rng = np.random.RandomState(42)
+        fit = OsipiBase(algorithm=args.algorithm)
+        S0 = 1
+        gd = GenerateData(rng=rng)
+        D = data["D"]
+        f = data["f"]
+        Dp = data["Dp"]  
+        # signal = gd.ivim_signal(D, Dp, f, S0, bvals, SNR, rician_noise)
+
         # Passing the values to the selectect algorithm and saving it
+        [f_fit, Dp_fit, D_fit] = fit.osipi_fit(signal, bvals)
```

Comment:
@IvanARashid had a preliminary wrapper that would make this work, but perhaps until that's ready we can loop over the point. Like this.

```
def loop_over_first_n_minus_1_dimensions(arr):
  """
  Loops over the first n-1 dimensions of a numpy array.

  Args:
      arr: A numpy array.

  Yields:
      A tuple containing the indices for the current iteration and a flattened view of the remaining dimensions.
  """
  n = arr.ndim
  for idx in np.ndindex(*arr.shape[:n-1]):
    flat_view = arr[idx].flatten()
    yield idx, flat_view


f_image = np.zeros_like(data.shape[:data.ndim-1]
D_image = np.zeros_like(data.shape[:data.ndim-1]
Dp_image = np.zeros_like(data.shape[:data.ndim-1]
for idx, view in loop_over_first_n_minus_1_dimensions(data):
    [f_fit, Dp_fit, D_fit] = fit.osipi_fit(view, bvals)
    f_image[idx]=f_fit
    Dp_image[idx]=Dp_fit
    D_image[idx]=D_fit
```

Reply:
Hello, how does this come in ?

Reply:
This shows how to loop over the data. `data` being loaded from the 4D image. The last dimension is the signal to fit.

Reply:
Also, how do I pass the value for the signal in `[f_fit, Dp_fit, D_fit] = fit.osipi_fit(signal, bvals)`

Reply:
Got it, thank you.

Reply:
> @IvanARashid had a preliminary wrapper that would make this work, but perhaps until that's ready we can loop over the point. Like this.
> 
> ```
> def loop_over_first_n_minus_1_dimensions(arr):
>   """
>   Loops over the first n-1 dimensions of a numpy array.
> 
>   Args:
>       arr: A numpy array.
> 
>   Yields:
>       A tuple containing the indices for the current iteration and a flattened view of the remaining dimensions.
>   """
>   n = arr.ndim
>   for idx in np.ndindex(*arr.shape[:n-1]):
>     flat_view = arr[idx].flatten()
>     yield idx, flat_view
> 
> 
> f_image = np.zeros_like(data.shape[:data.ndim-1]
> D_image = np.zeros_like(data.shape[:data.ndim-1]
> Dp_image = np.zeros_like(data.shape[:data.ndim-1]
> for idx, view in loop_over_first_n_minus_1_dimensions(data):
>     [f_fit, Dp_fit, D_fit] = fit.osipi_fit(view, bvals)
>     f_image[idx]=f_fit
>     Dp_image[idx]=Dp_fit
>     D_image[idx]=D_fit
> ```
I was able to make it run by using the np.getfromtxt as you made mention and also using another approach from this but similar.